homeassistant/components/device_tracker/__init__.py

```diff
@@ -207,18 +207,14 @@ def async_see_service(call):
         args = {key: value for key, value in call.data.items() if key in
                 (ATTR_MAC, ATTR_DEV_ID, ATTR_HOST_NAME, ATTR_LOCATION_NAME,
                  ATTR_GPS, ATTR_GPS_ACCURACY, ATTR_BATTERY, ATTR_ATTRIBUTES,
-                 ATTR_SOURCE_TYPE)}
-
-        try:
-            consider_home = call.data[ATTR_CONSIDER_HOME]
-        except KeyError:
-            pass
-        else:
-            args[ATTR_CONSIDER_HOME] = timedelta(seconds=int(consider_home))
+                 ATTR_SOURCE_TYPE, ATTR_CONSIDER_HOME)}
 
         yield from tracker.async_see(**args)
 
-    hass.services.async_register(DOMAIN, SERVICE_SEE, async_see_service)
+    hass.services.async_register(
+        DOMAIN, SERVICE_SEE, async_see_service,
+        schema=vol.Schema({ATTR_CONSIDER_HOME: cv.time_period},
```

Comment:
Please define the schema as a constant at the module level. Also please validate all possible service attributes in the schema. The schema should not allow extra.

Reply:
Previously none of the arguments were validated, so this just enforces validation on only `ATTR_CONSIDER_HOME`, while chaning nothing else. But yeah, I agree, I initially wanted to do what you said, but did this to keep changes minimal. 
I'll add a general schema for the service.